packages/evolution-backend/src/services/routing/RouteCalculationFromTransition.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+// This file contains route calculation utilities that can be used by the
+// survey. The should call chaire-lib's routing services to calculate the route.
+import { RoutingOrTransitMode } from 'chaire-lib-common/lib/config/routingModes';
+import { RouteCalculationParameter, RoutingResultByMode } from './types';
+import projectConfig from '../../config/projectConfig';
+import { URL } from 'url';
+
+// TODO Handle token expiration. For now though, the evolution app will restart more often than the token expiration
+let token: string | undefined = undefined;
+
+const getTransitionUrl = (transitionUrl: string) =>
+    new URL(transitionUrl.startsWith('http') ? transitionUrl : `http://${transitionUrl}`);
+
+const getToken = async () => {
+    if (token !== undefined) {
+        return token;
+    }
+
+    const transitionUrlObj = getTransitionUrl(projectConfig.transitionApi!.url);
+    transitionUrlObj.pathname = '/token';
+    const response = await fetch(transitionUrlObj.toString(), {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+            usernameOrEmail: projectConfig.transitionApi!.username,
+            password: projectConfig.transitionApi!.password
+        })
+    });
+
+    // FIXME Every request will try to get a token, even if the previous one failed. Should we cache the error and not retry? Or retry only after x time?
```

Comment:
We could have some form of exponential backoff mecanism for the retry. But at the amount of request that we expect for now, that's not really a problem. But we should make sure the error is logged (I don't know if the throw there will be loggued.

Reply:
It will be logged by the main function. If we log it here, it will be logged twice

Reply:
Added the exponential backoff in the comment